pkg/deploy/tls/certificates.go

```diff
@@ -0,0 +1,174 @@
+//
+// Copyright (c) 2012-2021 Red Hat, Inc.
+// This program and the accompanying materials are made
+// available under the terms of the Eclipse Public License 2.0
+// which is available at https://www.eclipse.org/legal/epl-2.0/
+//
+// SPDX-License-Identifier: EPL-2.0
+//
+// Contributors:
+//   Red Hat, Inc. - initial API and implementation
+//
+
+package tls
+
+import (
+	"context"
+	"reflect"
+	"strings"
+
+	"github.com/eclipse-che/che-operator/pkg/deploy"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	injector = "config.openshift.io/inject-trusted-cabundle"
+	// CheCACertsConfigMapLabelKey is the label value which marks config map with additional CA certificates
+	CheCACertsConfigMapLabelValue = "ca-bundle"
+	// CheAllCACertsConfigMapName is the name of config map which contains all additional trusted by Che TLS CA certificates
+	CheAllCACertsConfigMapName = "ca-certs-merged"
+	// CheMergedCAConfigMapRevisionsAnnotationKey is annotation name which holds versions of included config maps in format: cm-name1=ver1,cm-name2=ver2
+	CheMergedCAConfigMapRevisionsAnnotationKey = "che.eclipse.org/included-configmaps"
+
+	// Local constants
+	// labelEqualSign consyant is used as a replacement for '=' symbol in labels because '=' is not allowed there
+	labelEqualSign = "-"
+	// labelCommaSign consyant is used as a replacement for ',' symbol in labels because ',' is not allowed there
+	labelCommaSign = "."
+)
+
+type Certificates struct {
+}
+
+func NewCertificates() *Certificates {
+	return &Certificates{}
+}
+
+func (c *Certificates) Reconcile(ctx *deploy.DeployContext) (reconcile.Result, bool, error) {
+	if ctx.Proxy.TrustedCAMapName != "" {
+		if ctx.CheCluster.Spec.Server.ServerTrustStoreConfigMapName == "" {
```

Comment:
How will `Proxy.TrustedCAMapName` be handled if `CheCluster.Spec.Server.ServerTrustStoreConfigMapName` is not empty?